### ISSUE TITLE
duplicate storage entry shown for the microSD card due to the presence of the link

### ISSUE THREAD

**Issue by unclejack** [ValveSoftware/SteamOS#1043]
#### Your system information

* Steam client version: 1684372665
* SteamOS version: 20230516.1000
* Opted into Steam client beta?: [Yes/No] Yes  
* Opted into SteamOS beta?: [Yes/No] Yes - main branch
* Have you checked for updates in Settings > System?: [Yes/No] Yes, today

#### Please describe your issue in as much detail as possible:
What I've done:
I installed all updates today - SteamOS and Steam client updates through the UI.
What I expected:
I expected to see two storage locations - one for the internal SSD and one for the microSD card.
What happened:
My microSD card has two mounts in the storage UI: one at /run/media/mmcblk0p1 and one at /run/media/deck/-UUID-
Two locations are shown in the install game installation UI as well. 

The games were also not showing properly in the storage UI for both storage locations. One of the two is also left as the default installation library.

This seems to be caused by the presence of a link. The link is from /run/media/mmcblk0p1 to the actual location of the mount - the one with the UUID. I guess this was put in place for compatibility with previous versions. It just needs to be handled in the Steam client in a better way.

#### Steps for reproducing this issue:

1. Make sure there's a microSD with games installed on it
2. Install all updates while using the main branch and Steam client beta
2. The microSD card may have two entries in all the relevant storage related UIs, instead of one

**Comment by unclejack** [comment]
I was able to sort this out in the end. The duplicate steam library storage vanished after doing the following:
1. shut the Steam Deck down
2. started the Steam Deck again
3. ejected the microSD from the library which had the eject option (the one with the UUID, the real mount)
4. shut down the Steam Deck
5. turned the Steam Deck
6. inserted the microSD card again
7. rebooted
8. only one library was left

When the microSD card was unmounted the last time, there wasn't a second duplicate library anymore.

Perhaps this is a SteamOS 3.5 issue.

**Comment by bertogg** [comment]
I have heard about this although I cannot reproduce it myself. I have doubts that this is due to the presence of the link however.

**Comment by unclejack** [comment]
I haven't done anything special to trigger this. It was just something I've run into on SteamOS 3.5 image from main with the beta. It might be something which can be reproduced.

I've reported this issue to make sure it's reported and fixed if possible.

**Comment by bertogg** [comment]
Yes, you did the right thing by reporting it. There have been some changes to UDisks that might be related to this, I'll tell you when they are available so you can test them. Thanks!

**Comment by bertogg** [comment]
All images from the main channel since `20230612.1000` come with the fixed version of UDisks, tell me if you are still having problems.

**Comment by unclejack** [comment]
I've tried to reproduce this bug by ejecting the microSD card and inserting it again. I couldn't make it show up two entries for the microSD card. I couldn't get it to show twice even if I booted without it and inserted it after booting into SteamOS.

It's very likely that this has been solved. It wasn't that difficult to trigger the bug before.

I'll close the issue. If you encounter this, you might want to open a ticket and explain exactly what you did to trigger it.
